docs/audio.rst

```diff
@@ -215,6 +244,153 @@ Sound Effects Example
 .. include:: ../examples/soundeffects.py
     :code: python
 
+
+AudioRecording & AudioTrack **V2**
+==================================
+
+There are two classes that can contain (or point to) audio data 
+and an associated sampling rate:
+
+- ``AudioRecording`` contains its own buffer, it's initialised with a size
+  defined in time, and it's the object type that ``microphone.record()``
+  returns.
+- ``AudioTrack`` does not hold its own buffer and instead points to a buffer
+  externally created. This buffer could be an ``AudioRecording``, or a basic
+  type like a ``bytearray``. It's similar to a
+  `memoryview <https://docs.micropython.org/en/v1.9.3/pyboard/reference/speed_python.html#arrays>`_
+  and it can be used to easily chop a portion of the audio data in the
+  ``AudioRecording`` or to modify its contents.
+
+AudioRecording
+--------------
+
+.. py:class::
+    AudioRecording(duration, rate=7812)
+
+    The ``AudioRecording`` object contains audio data and the sampling rate
+    associated to it.
+
+    The size of the internal buffer will depend on the ``rate``
+    (number of samples per second) and ``duration`` parameters.
+
+    :param duration: Indicates how many milliseconds of audio this
+        instance can store.
+    :param rate: The sampling rate at which data will be stored
+        via the microphone, or played via the ``audio.play()`` function.
+
+    .. py:function:: set_rate(sample_rate)
+
+        Configure the sampling rate associated with the data in the
+        ``AudioRecording`` instance.
+
+        :param sample_rate: The sample rate to set.
+
+    .. py:function:: get_rate()
+
+        Return the configured sampling rate for this
+        ``AudioRecording`` instance.
+
+        :return: The configured sample rate.
+
+    .. py:function:: copy()
```

Comment:
Do we need this method?  The Python way would be to use the constructor:
```py
new_recording = AudioRecording(old_recording)
```
(I can't remember if we discussed this or not.)

Reply:
Yes, we looked at picking between a copy constructor and a `copy()` method, in the end we decided to go with the method because the micro:bit API hasn't used copy constructors before and the `copy()` method was already a pattern used in classes like `Image`.

Reply:
OK, I've implemented `AudioRecording.copy()`.

(There's actually also the existing `SoundEffect.copy()`, so that's nice and consistent.)